### PR TITLE
Port 2.x 30-actions.sh to 3.0

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
**Description of the Pull Request (PR):**

Port 2.x 30-actions.sh to 3.0 

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [user](https://www.github.com/sylabs/singularity-userdocs) and [administrator](https://www.github.com/sylabs/singularity-admindocs) documentation bases.
- [x] I have tested this PR locally with a `make testall`
- [x] This PR is against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularity-maintainers

### CODE REVIEW

**Review by jscook2345 — COMMENTED** [review]
So some of this is duplicated in other tests. The actions test itself is non super specific. I would prefer breaking this out into each action. There already exists an 'exec_test'. Perhaps create a 'shell_test', and a 'run_test' to contain those parts.

**Comment by ArangoGutierrez** [comment]
@jscook2345 
> So some of this is duplicated in other tests. The actions test itself is non super specific. I would prefer breaking this out into each action.
```go 
func TestSingularityActions(t *testing.T) {}
```
is calling 3 local funcs, (non Upercased) , so they can share a single SIF for all the test.
I don't see duplicated test for actions, could you point me to the duplicated one, so I can quick fix on a commit.
```go
	// singularity run
	t.Run("run", testSingularityRun)
	// singularity exec
	t.Run("exec", testSingularityExec)
	// singularity shell
	t.Run("shell", testSingularityShell)
```

> There already exists an 'exec_test'. Perhaps create a 'shell_test', and a 'run_test' to contain those parts.

has been changed for `actions_test.go` to reflect cli tree, on which `actions.go` stores `run/shell/exec`

File: src/cmd/singularity/actions_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+type execOpts struct {
+	binds   []string
+	contain bool
+	home    string
+	workdir string
+	pwd     string
+}
+
+func imageExec(opts execOpts, imagePath string, command []string) ([]byte, error) {
```
Comment (bauerm97) [inline src/cmd/singularity/actions_test.go]: What is the purpose of this function if it never gets called?

File: src/cmd/singularity/actions_test.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+		{cmdPath, "NoCommand", []string{"run", imagePath}, 0},
+		{cmdPath, "true", []string{"run", imagePath, "true"}, 0},
+		{cmdPath, "false", []string{"run", imagePath, "false"}, 1},
+		// Testing run command properly hands arguments
+		{"sh", "arguments", []string{"-c", fmt.Sprintf("singularity run %s foo | grep foo", imagePath)}, 0},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			cmd := exec.Command(tt.binName, tt.argv...)
```
Comment (bauerm97) [inline src/cmd/singularity/actions_test.go]: If we've created a function to run an image, then why are we not using it?

**Review by bauerm97 — CHANGES_REQUESTED** [review]

File: src/cmd/singularity/actions_test.go (hunk 1 above)
Comment (ArangoGutierrez) [reply src/cmd/singularity/actions_test.go]: that func is used by `build_test.go` and `docker_test.go` , not in this file. as we need to catch the exit code

File: src/cmd/singularity/actions_test.go (hunk 2 above)
Comment (ArangoGutierrez) [reply src/cmd/singularity/actions_test.go]: `imageExec()` does not catch and return the exit code. should we modify that func to do it. or re allocate `imageExec()` to a diff test file

**Comment by ArangoGutierrez** [comment]
@bauerm97  func `imageExec()` has been updated to be general used in all test.

**Review by bauerm97 — APPROVED** [review]

**Pull request merged**